src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -3742,6 +3804,21 @@ int wm_vuldet_oval_xml_parser(OS_XML *xml, XML_NODE node, wm_vuldet_db *parsed_o
                     }
                     parsed_oval->vulnerabilities->ignore = ignore;
                     os_strdup(node[i]->values[j], parsed_oval->vulnerabilities->state_id);
+                    if (dist == FEED_SUSE && !parsed_oval->vulnerabilities->deps && parsed_oval->vulnerabilities->prev 
+                        && parsed_oval->vulnerabilities->prev->deps) {
+                        /*
+                        * The dependency list for SUSE vulnerabilities is only filled for the first package
+                        * of each <criteria operator="AND"> node.
+                        * This structure is then copied for subsequent vulnerabilities.
+                        */
+                        int elements = parsed_oval->vulnerabilities->prev->deps->elements;
+                        os_calloc(1, sizeof(dependencies), parsed_oval->vulnerabilities->deps);
+                        os_calloc(elements, sizeof(char *), parsed_oval->vulnerabilities->deps->test_ref);
+                        for (int i = 0; i < elements; i++) {
```

Comment:
Here we cannot reset the `i` value which is being used for the loop iterating the `node[i]` elements